Fix project and scm urls in child project POMs

lucifer4j opened the pull request:
Maven's default behaviour is to append path and artifact id to parent  project url for setting child project's url. But this is not helpful  because these urls, for instance https://github.com/quarkusio/quarkus/bom/quarkus-bom  end up in 404s. Similarly, the scm urls are broken. The right behaviour  would most likely be to point to just https://github.com/quarkusio/quarkus/.

Thankfully, Maven 3.6.1 introduced some attribute to set the child  project's url to the project url of the parent without modifications. Therefore, set the child.project.url.inherit.append.path project attribute and various scm attributes to ensure the correct urls are generated.

The fix can be verified by running `help:effective-pom` maven goal and looking at the output.

lucifer4j committed:
Fix project and scm urls in child project POMs

Maven's default behaviour is to append path and artifact id to parent
 project url for setting child project's url. But this is not helpful
 because these urls, for instance https://github.com/quarkusio/quarkus/bom/quarkus-bom
 end up in 404s. Similarly, the scm urls are broken. The right behaviour
 would most likely be to point to just https://github.com/quarkusio/quarkus/.

Thankfully, Maven 3.6.1 introduced some attribute to set the child
 project's url to the project url of the parent without modifications.
Therefore, set the child.project.url.inherit.append.path project attribute
and various scm attributes to ensure the correct urls are generated.

gastaldi reviewed (approved):
LGTM

gastaldi commented:
@lucifer4j do you have a reference link to share showing these properties?

lucifer4j commented:
@gastaldi Ah yes. Sorry forgot to put it here earlier. Here it is: [Maven POM schema reference](https://maven.apache.org/ref/3.8.4/maven-model/maven.html#:~:text=Description-,child.project.url.inherit.append.path,-String). Anchors are not available on the page so you may need to scroll a bit to find the attributes. These are the attributes to look for: `child.project.url.inherit.append.path`, `child.scm.connection.inherit.append.path`, `child.scm.developerConnection.inherit.append.path` and `child.scm.url.inherit.append.path`. 

There is also 1 additional attribute available `child.site.url.inherit.append.path`. I didn't see the site url to be specified anywhere in the parent pom, the pom in docs or in the quick search of the pom files in the repo so I didn't use that attribute in the PR.

famod reviewed (approved):
Can't say I have used those myself, but looks alright!

But I think `independent-projects` should be covered as well?

lucifer4j committed:
Fix project and scm urls in independent project POMs

Just like other child projects the project and scm urls are broken for
independent projects. However, these projects inherit from JBoss Parent
POM so we need to override the url once in each parent independent
project as well in addition to the adding the inherit.append.path
attributes.

lucifer4j commented:
Hi! Sorry for the delay in replying. @famod I have updated independent projects as well. Since the independent projects inherit from JBoss Parent POM, I also needed to add url and scm at a few places in the parent independent projects. Otherwise, the url would have been set to the jboss repo.

lucifer4j commented:
I realised a few minutes ago that the jboss-parent-pom is probably a better place to fix this issue so I have opened a PR there. If that PR gets accepted, this one can probably be discarded.

famod commented:
> I realised a few minutes ago that the jboss-parent-pom is probably a better place to fix this issue so I have opened a PR there. If that PR gets accepted, this one can probably be discarded.

Thanks! But I think we should merge this regardless as things are moving very slow for jboss-parent.

gastaldi commented:
Oops, clicked the Update Branch by accident, should have chosen Rebase

lucifer4j commented:
@famod Oh! Makes sense :+1:
